plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+When a commitment is marked as `ConfirmedValid`, we can finalize portions of the L2 chain. The exact finalization algorithm is described [here](#Finalization)
+
+The DA server MUST look at the challenge contract on L1 to find the data from the commitment when it is resolved with an Optimistic bridge.
+The DA server MAY look at the challenge contract on L1 to find the data from the commitment when it is resolved with an ZK bridge; however, it is a breaking change to switch from not looking at the challenge contract to looking at the challenge contract.
+
+By only looking at an opaque commitment and letting the DA server be responsible for fetching the data & letting the DA Challenge contract be responsible for validating the data, no parts of the monorepo need to be modified to work with a new DA Layer.
+
+
+### Alternatives
+
+We initially considered the following options for designing a DA challenge. These where rejected because they are too DA layer specific inside the op-node.
+* The op-node does verification of data (no bond payouts / commitment validity checks on L1)
+* Modify the existing DAC bridge to work with different commitments.
+* Integrating a ZK Bridge (for example Blobstream).
+
+We did consider the following as legitimate generic options.
+
+We could combine the V1 and V1 SC events into a single interface. This would mean that there is only one adapter from L1; however it would mix together bridges of different types and increase implementation complexity.
+
+We also could directly read SC events in the derivation pipeline (and we do this at the moment); however, it requires mixing the V1 and V2 interfaces into. In addition converting to the status events works better inside the pipeline & enables future optimizations.
+
+
+### Future Work
+
+In the future, we can remove the SC adapter & the DA server can become responsible for providing the status information.
+The DA server would be able to return `ConfirmedValid` or `ConfirmedInvalid` when it knows that a transaction on L1 would result in the same event.
+The benefit of L1 execution for `ConfirmedValid` is that we will be able to significantly reduce the finalization delay for DA Layers that have timely ZK bridges.
+
+## Finalization
+
+At a high level, finalization is process of marking L2 blocks as not being able to be be reorged.
+If a commitment is marked as invalid, we have to reorg the L2 chain, therfore finality in Plasma Mode is about tracking when commitments finalize.
+
+To implement finality, we track commitments (even skipped or invalid commitments in order). When the derivation pipeline has advanced to the point where
+the commitment status changes from `OptimisticOK` to `ConfirmedValid` or `ConfirmedInvalid`, we record the L1 block in which that occurred.
+We then wait for that L1 block to be finalized. Once that L1 block has been finalized, the commitment status can never change.
```

Comment:
And to confirm, OptimisticOK turns into Confirmed when:
* In Optimistic, when the challenge window expires
* In ZK, when a proof is supplied

Reply:
In ZK it could also be if the challenge period expires. Blobstream is half ZK where you could immediately get `Valid/Invalid` but don't want to call those contracts unless there is a dispute